### PR TITLE
SDL: Split the OpenGL code to a separate GraphicsManager

### PR DESCRIPTION
**Pull request by bgK**
The general goal for these changes is to move the OpenGL specific code out of SurfaceSdlGraphicsManager so that the code is less messy, and it is easier to add features. Possibly in the future the OpenGL code could be further separated from the SDL code so that it can be reused for other platforms.

A new class, ResVMSdlGraphicsManager has been added in the class hierarchy between SurfaceSdlGraphicsManager and SdlGraphicsManager to be able to share common code between the surface and OpenGL graphics managers. The OpenGL code has been moved to OpenGLSdlGraphicsManager, subclassing ResVMSdlGraphicsManager.

The SDL platform code now chooses to instantiate either SurfaceSdlGraphicsManager or OpenGLSdlGraphicsManager based on the "accel3d" parameter of the "setupScreen" method.

The OpenGL code used to draw the overlay has been extracted to a new class named "TiledSurface". This new class has slightly better dirty texture handling which could improve performance a bit.

The way the overlay is drawn when in OpenGL mode was changed. There are now two 2D surfaces being rendered instead of one. The first one contains a capture of the game screen. The second one is drawn on top of it with alpha blending enabled and contains the actual overlay / GMM. This finally fixes the glitch where GUI dialogs would not disappear when closed in the GMM, at the cost of additional VRAM usage.

### CODE REVIEW

**Comment by Botje** [comment]
Nice refactor! 👍

**Comment by bgK** [comment]
I'll merge this in a few days unless somebody wants to further review it.

**Comment by aquadran** [comment]
I want review it :)

File: backends/graphics/sdl/resvm-sdl-graphics.cpp [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+static const OSystem::GraphicsMode s_supportedGraphicsModes[] = {
+		{0, 0, 0}
+};
+
+ResVmSdlGraphicsManager::ResVmSdlGraphicsManager(SdlEventSource *source, SdlWindow *window, const Capabilities &capabilities) :
+		SdlGraphicsManager(source, window),
+		_fullscreen(false),
+		_lockAspectRatio(true),
+		_overlayVisible(false),
+        _overlayWidth(0),
```
Comment (aquadran) [inline backends/graphics/sdl/resvm-sdl-graphics.cpp]: formatting

**Comment by aquadran** [comment]
:+1:

File: backends/graphics/sdl/resvm-sdl-graphics.cpp (hunk 1 above)
Comment (bgK) [inline backends/graphics/sdl/resvm-sdl-graphics.cpp]: fixed

**Pull request merged**